docs/spec_v5.md

```diff
@@ -151,30 +148,91 @@ Content-Length: 27
 {"message":"Hello, World!"}
 ```
 
-## Errors
 
-Twirp error responses are always JSON-encoded, regardless of
-the request's Content-Type, with a corresponding
-`Content-Type: application/json` header. This ensures that
-the errors are human-readable in any setting.
+## Errors
 
-Twirp errors are a JSON object with three keys:
+Erros are non-200 JSON responses with the keys:
 
-* **code**: One of the Twirp error codes as a string.
-* **msg**: A human-readable message describing the error
-  as a string.
-* **meta**: An object with string keys and values holding
-  arbitrary additional metadata describing the error.
+* **code**: (string) One of the valid Twirp error codes.
+* **msg**: (string) Human-readable message describing the error.
+* **meta**: (optional object) arbitrary additional metadata 
+  describing the error. Keys and values must be strings.
 
 Example:
+
+```json
+{
+  "code": "permission_denied",
+  "msg": "thou shall not pass",
+  "meta": {
+    "target": "Balrog"
+  }
+}
 ```
+
+### Error Responses are JSON
+
+Twirp error responses are always JSON-encoded, the response Content-Type
+header is always `application/json`, regardless of the request's 
+Content-Type (protbuf requests also respond with json errors).
+This ensures that errors are human-readable and easy to parse by intermediary proxies.
+
+Example Error Response:
+
+```
+HTTP/1.1 500 Internal Server Error
+Content-Type: application/json
+Content-Length: 48
+
+{"code":"internal","msg":"Something went wrong"}
+```
+
+### Twirp Error Codes
+
+The possible values for the Error "code" are intentionaly similar to
```

Comment:
Revise this section's text to this:

"Twirp errors always include an error code. This code is represented as a string and must be one of a fixed set of codes, listed in the table below. Each code has an associated HTTP Status Code. When a server responds with the given error code, it must set the corresponding HTTP Status Code for the response."